tinygrad/tensor.py

```diff
@@ -2000,20 +2000,21 @@ def _pool(self, k_:Tuple[sint, ...], stride:Union[Tuple[int, ...], int]=1, dilat
     return x.permute(*range(len(noop)), *[len(noop)+i*2 for i in range(len(i_))], *[len(noop)+i*2+1 for i in range(len(i_))])
 
   def _padding2d(self, padding:Union[int, Sequence[int]], dims:int) -> Sequence[int]:
+    if not isinstance(padding, int) and not (len(padding) == 2*dims or len(padding) == dims):
+      raise ValueError(f"Expected padding of length {2*dims} or {dims}, but got {len(padding)} for tensor of shape {self.shape}")
```

Comment:
I moved [this](https://github.com/tinygrad/tinygrad/pull/8109/files#diff-7703e7b63c9844f019ae3609629fd9c9bab290d11ae6608ed2d09edd49bab2dbL2096) from `conv2d` into `_padding2d`

should we rename `_padding2d` to something like `_resolve_pool_pad`? And run it in the beginning of every pool function to resolve the pad like we do with `_resolve_dim`?